### PR TITLE
Stops runtime in cult spirit realm rune when observer doesn't have a mind

### PR DESCRIPTION
**Pull request by chuga-git**
## What Does This PR Do

See title. This will shortcircuit and continue before the HAS_TRAIT check can runtime as well. Observers having valid clients but a null mind is really funky and should probably be investigated further.

## Why It's Good For The Game

Fixes edge case runtime.
## Testing

I can't test this locally (or reproduce the original issue), but it compiles. 
<hr>

### Declaration
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

<hr>

### CODE REVIEW

File: code/game/gamemodes/cult/runes.dm [hunk 1]
```diff
@@ -940,7 +940,7 @@ structure_check() searches for nearby cultist structures required for the invoca
 			continue
 		if(!HAS_TRAIT(O, TRAIT_RESPAWNABLE) || QDELETED(src) || QDELETED(O))
 			continue
-		if(O.mind.current && HAS_TRAIT(O.mind.current, SCRYING))
+		if(O?.mind?.current && HAS_TRAIT(O.mind.current, SCRYING))
```
Comment (Contrabang) [inline code/game/gamemodes/cult/runes.dm:943]: This means that the mindless ghost will be added to the rune.
```suggestion
		if(!O.mind)
				continue
		if(O.mind.current && HAS_TRAIT(O.mind.current, SCRYING))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/cult/runes.dm (hunk 1 above)
Comment (chuga-git) [reply code/game/gamemodes/cult/runes.dm:943]: fuck

File: code/game/gamemodes/cult/runes.dm [hunk 2]
```diff
@@ -940,7 +940,9 @@ structure_check() searches for nearby cultist structures required for the invoca
 			continue
 		if(!HAS_TRAIT(O, TRAIT_RESPAWNABLE) || QDELETED(src) || QDELETED(O))
 			continue
-		if(O.mind.current && HAS_TRAIT(O.mind.current, SCRYING))
+		if(!O.mind || !O.mind.current)
+			continue
+		if(HAS_TRAIT(O.mind.current, SCRYING))
```
Comment (Contrabang) [inline code/game/gamemodes/cult/runes.dm:945]: I'm dumb, we still want this to succeed if they have a mind but no current body
```suggestion
		if(!O.mind)
			continue
		if(O.mind.current && HAS_TRAIT(O.mind.current, SCRYING))
```

**Review by FunnyMan3595 — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]
